packages/airview-cms/README.md

```diff
@@ -1 +1,395 @@
+
+### String Widget
+
+The string widget is used to capture a string input from a user (not multiline input). You can include the string widget within your collection field by passing `string` as the value to the `widget` key.
+
+- **Name:** `string`
+- **UI:** Text input
+- **Data type:** string
+- **Options:**
+- - `label`: string - A human readable label for the UI - _required_
+- - `placeholder`: string - An optional placeholder value for when the text area has no input value from a user
+
+**Example:**
+
+```javascript
+{
+  label: "Price",
+  name: "price",
+  widget: "string",
+  placeholder: "Enter a price for this product"
+}
+```
+
+### Boolean Widget
+
+The boolean widget is used to capture a boolean value from a user. You can include the boolean widget within your collection field by passing `boolean` as the value to the `widget` key.
+
+- **Name:** `boolean`
+- **UI:** Checkbox input
+- **Data type:** boolean
+- **Options:**
+- - `label`: string - A human readable label for the UI - _required_
+- - `defaultValue`: boolean - Sets the default value for the UI - _required_
+
+**Example:**
+
+```javascript
+{
+  label: "In Stock",
+  name: "in_stock",
+  defaultValue: false,
+  widget: "boolean",
+}
+```
+
+### Date Widget
+
+The date widget is used to capture a date from a user. You can include the date widget within your collection field by passing `date` as the value to the `widget` key.
+
+- **Name:** `date`
+- **UI:** Date select input
+- **Data type:** ISO date string
+- **Options:**
+- - `label`: string - A human readable label for the UI - _required_
+- - `defaultValue`: ISO date string - Sets the default value for the UI - _required_
+- - `format`: string - Formats the display of the date to the user, pass a valid [Day.js format string](https://day.js.org/docs/en/display/format). Defaults to `DD/MM/YYYY`
+- - `minDate`: ISO date string - Sets an optional minimum selectable date.
+- - `maxDate`: ISO date string - Sets an optional maximum selectable date.
+
+**Example:**
+
+```javascript
+{
+  label: "Available From",
+  name: "available_from",
+  widget: "date",
+  minDate: "2022-05-01T00:00:00Z",
+  maxDate: "2022-05-31T00:00:00Z",
+  defaultValue: "2022-05-01T00:00:00Z",
+  format: "DD/MM/YY"
+}
+```
+
+### Entry Select Widget
+
+The entry select widget is used to select an entry from a specific collection. You can include the entry select widget within your collection field by passing `entrySelect` as the value to the `widget` key.
+
+- **Name:** `entrySelect`
+- **UI:** Select Picker UI
+- **Data type:** string (entryID)
+- **Options:**
+- - `label`: string - A human readable label for the UI - _required_
+- - `collection`: string - The specific collection key, ued to populate the select picker with the relevant entries for the passed collection - _required_
+
+**Example:**
+
+```javascript
+{
+  label: "Related Coffee",
+  name: "related_coffee",
+  widget: "entrySelect",
+  collection: "coffee"
+}
+```
+
+## CMS Context
```

Comment:
Does JS/React have a way of documenting the below hooks in code for hinting in the IDE? The documentation is good, but it'd be helpful to have it with the code. I know we can generate this type of documentation from code in other languages. Given time constraints, we should probably get this merged the raise a task to cover moving it into code later.

Reply:
Good question, I'm not sure to be honest. 

Moving to TypeScript would help, but that's a big task; JSDoc could work, failing that we could inline an example in the ReadMe.

Added a ticket to track:

https://github.com/AirWalk-Digital/airview-issues/issues/259